component/src/main/java/com/synopsys/integration/alert/component/settings/proxy/web/SettingsProxyController.java

```diff
@@ -0,0 +1,61 @@
+package com.synopsys.integration.alert.component.settings.proxy.web;
+
+import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.synopsys.integration.alert.common.rest.AlertRestConstants;
+import com.synopsys.integration.alert.common.rest.ResponseFactory;
+import com.synopsys.integration.alert.common.rest.api.ReadPageController;
+import com.synopsys.integration.alert.common.rest.api.StaticConfigResourceController;
+import com.synopsys.integration.alert.common.rest.api.ValidateController;
+import com.synopsys.integration.alert.common.rest.model.AlertPagedModel;
+import com.synopsys.integration.alert.common.rest.model.ValidationResponseModel;
+import com.synopsys.integration.alert.component.settings.proxy.action.SettingsProxyCrudActions;
+import com.synopsys.integration.alert.component.settings.proxy.action.SettingsProxyValidationAction;
+import com.synopsys.integration.alert.component.settings.proxy.model.SettingsProxyModel;
+
+@RestController
+@RequestMapping(AlertRestConstants.SETTINGS_PROXY_PATH)
+public class SettingsProxyController implements StaticConfigResourceController<SettingsProxyModel>, ValidateController<SettingsProxyModel>, ReadPageController<AlertPagedModel<SettingsProxyModel>> {
+    private final SettingsProxyCrudActions configActions;
+    private final SettingsProxyValidationAction validationAction;
+
+    @Autowired
+    public SettingsProxyController(SettingsProxyCrudActions configActions, SettingsProxyValidationAction validationAction) {
+        this.configActions = configActions;
+        this.validationAction = validationAction;
+    }
+
+    @Override
+    public SettingsProxyModel create(SettingsProxyModel resource) {
+        return ResponseFactory.createContentResponseFromAction(configActions.create(resource));
+    }
+
+    @Override
+    public SettingsProxyModel getOne(UUID id) {
+        return ResponseFactory.createContentResponseFromAction(configActions.getOne(id));
+    }
+
+    @Override
+    public AlertPagedModel<SettingsProxyModel> getPage(Integer pageNumber, Integer pageSize, String searchTerm) {
+        return ResponseFactory.createContentResponseFromAction(configActions.getPaged(pageNumber, pageSize));
+    }
```

Comment:
Do we plan on supporting pages of proxy info? If not, we should do the getOne without an ID like we did for encryption

Reply:
When we were creating the controller we discussed future Alert changes where we could have multiple proxies configured and the user could pick and choose which one to apply. We decided that we could implement and support pages via the API now but have it unused for now within the UI.